lib/web_ui/lib/src/engine/view_embedder/embedding_strategy/full_page_embedding_strategy.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:ui/src/engine/dom.dart';
+import 'package:ui/src/engine/util.dart' show assertionsEnabled, setElementStyle;
+
+import 'embedding_strategy.dart';
+
+class FullPageEmbeddingStrategy extends EmbeddingStrategy {
+  @override
+  void initialize({
+    Map<String, String>? embedderMetadata,
+  }) {
+    // ignore:avoid_function_literals_in_foreach_calls
+    embedderMetadata?.entries.forEach((MapEntry<String, String> entry) {
+      _setHostAttribute(entry.key, entry.value);
+    });
+    _setHostAttribute('flt-embedding', 'full-page');
+
+    _applyViewportMeta();
+    _setHostStyles();
+  }
+
+  @override
+  void attachGlassPane(DomElement glassPaneElement) {
+    /// Tweaks style so the glassPane works well with the hostElement.
+    glassPaneElement.style
+      ..position = 'absolute'
+      ..top = '0'
+      ..right = '0'
+      ..bottom = '0'
+      ..left = '0';
```

Comment:
Ah, this is where I got that intuition from 😆

Reply:
This only works with `position: absolute`.

Eventually we should make both implementations converge, and make the difference between full screen and single element just a little bit of extra CSS :)